src/object/rpc_csum.c

```diff
@@ -196,6 +192,8 @@ crt_proc_struct_dcs_iod_csums_adv(crt_proc_t proc, crt_proc_op_t proc_op,
 	if (DECODING(proc_op)) {
 		PROC(uint32_t, &iod_csum->ic_nr);
 		D_ALLOC_ARRAY(iod_csum->ic_data, iod_csum->ic_nr);
+		if (iod_csum->ic_data == NULL)
+			return -DER_NOMEM;
```

Comment:
This would have resulted in a segfault here before presumably?

Reply:
Yes, if memory failed to be allocated then a segfault would have probably occurred when ic_data was attempted to be accessed.

Reply:
This warrants it's own ticket and a 1.2 backport then I assume?

Reply:
Yes, probably would have, and I debated this with myself, but I ended up lumping all the fixes for the issues reported by NLT into one patch. I will work on the backport.